mne/io/eeglab/eeglab.py

```diff
@@ -355,6 +351,22 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
         _read_segments_file(self, data, idx, fi, start, stop, cals, mult,
                             dtype=np.float32, n_channels=self.info['nchan'])
 
+    # XXX: to be removed when deprecating montage
+    def set_montage(self, montage, set_dig=True, update_ch_names=False,
+                    verbose=None):
+        """To be removed."""  # noqa
+        cal = set([ch['cal'] for ch in self.info['chs']]).pop()
+        coord_frame = set([ch['coord_frame'] for ch in self.info['chs']]).pop()
+        super(RawEEGLAB, self).set_montage(montage, set_dig=set_dig,
+                                           update_ch_names=update_ch_names,
+                                           verbose=verbose)
+        # Revert update_ch_names modifications in cal and coord_frame
+        if update_ch_names:
+            for ch in self.info['chs']:
+                ch['cal'] = cal
+                ch['coord_frame'] = coord_frame
```

Comment:
For example, it is probably a bug that the main `set_montage` function currently does not set `ch['coord_frame']` (regardless of the `update_ch_names` value) and it's something that we should have the main `set_montage` function do

Reply:
yes we discussed this with Sik this morning. set_montage when a montage is passed should update the chs to head coord system...

Reply:
what do you suggest @larsoner ?

Reply:
agreed.

But all this code would be removed anyway by something like this:

```py
if isinstace(foo, str):
   warn
   return _create_digitization_from_template_montage(foo)
elif instance(foo, Montage):
   warn
   return _create_digitization_from_montage(foo)
elif instance(foo, DigMontage):
   warn
   return _create_digitization_from_DigMontage(foo)
else:
   pass
```

So I'm not sure its worth fixing.